src/Sparrow/Utils/NativeMemory.cs

```diff
@@ -124,13 +124,30 @@ public static void Free(byte* ptr, long size, ThreadStats stats)
 
         public static void Free(byte* ptr, long size)
         {
-            Free(ptr, size, ThreadAllocations.Value);
+            Debug.Assert(ptr != null);
+            Interlocked.Add(ref _totalAllocatedMemory, -size);
+            Marshal.FreeHGlobal((IntPtr)ptr);
```

Comment:
Shouldn't `public static void Free(byte* ptr, long size, ThreadStats stats)` call this method to avoid code duplication?